packages/host/app/services/loader-service.ts

```diff
@@ -98,11 +98,29 @@ export default class LoaderService extends Service {
     } else {
       body = await request.text();
     }
-    return await this.loader.fetch(request.url, {
+    let response = await this.loader.fetch(request.url, {
       method: request.method,
       headers: new Headers(request.headers),
       body,
     });
+
+    // Obtaining a new JWT
+    // if receiving `Authorization` error,
+    // which means user permissions is updated from the server.
+    if (!response.ok && response.status === 403) {
+      await realmSession.refreshToken();
+      if (!realmSession.rawRealmToken) {
+        return null;
+      }
+      request.headers.set('Authorization', realmSession.rawRealmToken);
+      response = await this.loader.fetch(request.url, {
```

Comment:
in my (still draft) [PR](https://github.com/cardstack/boxel/pull/1049/files#diff-b6059baf94cff52ea5df0ab5f23724bc3c52a3685ea3a844fc1d9c9c84192bddR459) where a handler with the same name was introduced for realm2realm auth, we explicitly made it so that the only responsibility of this middleware is to add the `Authorization` header to the request object. Perhaps here it should be the same as it's much easier to reason about and reduce chance of bugs and overhead by reducing internal fetches in the url handlers as much as possible

Reply:
It makes sense if this handler only including the token to `Authorization` header. While in this PR, this handler also have to deal with response error, so I think it's fine to call fetch in this handler.